### ISSUE TITLE
Shortcuts/hotkeys to move main area widgets between areas

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#11373]
### Problem

- My mouse has an issue with the click events being interrupted; it is a physical defect but I am keeping it for now. I believe users with dyskinesia will be similarly affected. The current mouse-only accessibility makes it very hard to arrange tabs for such users.
![annoyance](https://user-images.githubusercontent.com/5832902/139439639-41f8264e-0185-462b-938c-e2bc75f7b43b.gif)
- Even when I had a fully functional mouse, moving tabs was still annoying and sometimes my finger could slip of; having the tab land in the area different that I wanted always resulted in frustration because re-layout takes a long time (3-10 seconds) and next attempt may or may not work
- Automated visual testing would benefit from having commands to re-arrange layout quickly

### Proposed Solution

I would like to have a set of commands, roughly "move to neighbouring area left/right/bottom/top" possibly with IDs like `move-to-area-left` for simplicity:

- I would like to be able to assign shortcuts for those, like:
   - `some modifier` + `right`: moves tab to the area right from the current one or create one if needed
- The commands should be accessible from the tab context menu as a sub-menu

![Screenshot from 2021-10-29 17-42-55](https://user-images.githubusercontent.com/5832902/139472197-a0de238c-343c-484d-9f13-983cd774cbc5.png)

### Additional context

Many window managers allow to arrange windows using <kbd>Alt</kbd> + arrows or other hotkeys.

- PyCharm has `Split Down/Right` and `Split And Move Down/Right`
- VS Code has `Split (Left/Right/Bottom/Top)`:

![Screenshot from 2021-10-29 17-44-09](https://user-images.githubusercontent.com/5832902/139472481-6e3adab7-fd36-4909-b593-3192cf65d639.png)

**Comment by EthanLowenthal** [comment]
Seems like a nice idea - I can take a crack at this

**Comment by Meriem-BenIsmail** [comment]
Hi, I'm looking into this.
